vault/expiration.go

```diff
@@ -1806,6 +1822,67 @@ func (m *ExpirationManager) deleteLockForLease(id string) {
 	m.lockPerLease.Delete(id)
 }
 
+func (m *ExpirationManager) sendEventByID(eventType logical.EventType, leaseID string) {
+	le, err := m.loadEntry(context.Background(), leaseID)
+	if err != nil {
+		m.logger.Warn("Error fetching lease information", "error", err)
+		return
+	}
+	m.sendEvent(eventType, le)
+}
+
+func (m *ExpirationManager) sendEvent(eventType logical.EventType, le *leaseEntry) {
```

Comment:
(This comment is all about understanding any further implications for a pre-existing issue, so is about discussion rather than suggesting any changes)

The `ExpirationManager` runs on every node that's not a DR secondary:

https://github.com/hashicorp/vault/blob/87376a1f5c676c9a54e1050b3e71e32c4ac92595/vault/core.go#L2392-L2398

However, clients of the events system are forwarded to the leader, so we aren't currently able to consume events from Perf secondaries, because they don't forward their events within the cluster. IIUC before this PR we would only be able to generate events on Perf secondaries if they have a local mount and thus accept writes to storage directly for that mount [citation needed from @ncabatoff]. Do lease events change the picture here at all?